proto/minder/v1/minder.proto

```diff
@@ -2559,70 +2559,73 @@ message ListEvaluationHistoryResponse {
-    EvaluationHistoryRule rule = 2;
+        // type is the entity type.
+        minder.v1.Entity type = 2;
 
-    // status contains the evaluation status.
-    EvaluationHistoryStatus status = 3;
+        // name is the entity name.
+        string name = 3;
+    }
 
-    // alert contains details of the alerts for this evaluation.
-    EvaluationHistoryAlert alert = 4;
+    message Rule {
+        // name is the name of the rule instance.
+        string name = 1;
 
-    // remediation contains details of the remediation for this evaluation.
-    EvaluationHistoryRemediation remediation = 5;
-}
+        // type is the name of the rule type.
+        string type = 2;
 
-message EvaluationHistoryEntity {
-    // id is the ID of the entity.
-    string id = 1;
+        // profile is the name of the profile which contains the rule.
+        string profile = 3;
+    }
 
-    // type is the entity type.
-    Entity type = 2;
+    message Status {
+        // status is one of (success, error, failure, skipped)
+        // not using enums to mirror the behaviour of the existing API contracts.
+        string status = 1;
 
-    // name is the entity name.
-    string name = 3;
-}
+        // details contains optional details about the evaluation.
+        // the structure and contents are rule type specific, and are subject to change.
+        string details = 2;
 
-message EvaluationHistoryRule {
-    // name is the name of the rule instance.
-    string name = 1;
+        // created_at is the timestamp of creation of this evaluation
+        google.protobuf.Timestamp evaluated_at = 3;
+    }
 
-    // type is the name of the rule type.
-    string type = 2;
+    message Remediation {
+        // status is one of (success, error, failure, skipped, not available)
+        // not using enums to mirror the behaviour of the existing API contracts.
+        string status = 1;
 
-    // profile is the name of the profile which contains the rule.
-    string profile = 3;
-}
+        // details contains optional details about the remediation.
+        // the structure and contents are remediation specific, and are subject to change.
+        string details = 2;
+    }
 
-message EvaluationHistoryStatus {
-    // status is one of (success, error, failure, skipped)
-    // not using enums to mirror the behaviour of the existing API contracts.
-    string status = 1;
+    message Alert {
+        // status is one of (on, off, error, skipped, not available)
+        // not using enums to mirror the behaviour of the existing API contracts.
+        string status = 1;
 
-    // details contains optional details about the evaluation.
-    // the structure and contents are rule type specific, and are subject to change.
-    string details = 2;
-}
+        // details contains optional details about the alert.
+        // the structure and contents are alert specific, and are subject to change.
+        string details = 2;
+    }
```

Comment:
The comments on `Remediation` also apply to `Alert`.  In fact, I'm wondering whether an `Alert` is a particular type of remediation.  @puerco @ethomson

Reply:
I think that is reflective of the current state of the universe, but not necessarily how things work in future.

One distinction which comes to mind is that a remediation generally operates against an entity, whereas an alert may not (for example, creating a Jira ticket does not involve changing the state of an entity).

Reply:
At this moment in time, they might. I am not so sure that they will remain the same as we evolve them in future.